Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedDataParameterDefinitionTest.cs

```diff
@@ -23,73 +23,98 @@
+  private static IEnumerable<(SqlMetaData[] columnMetaData, TestData[] testData)> GetTestCollections ()
+  {
+    yield return ([new SqlMetaData("Value", SqlDbType.UniqueIdentifier)], []);
+    yield return ([new SqlMetaData("Value", SqlDbType.UniqueIdentifier)],
+            [
+                new TestData { Item = new Guid("280D68FF-BEA9-47CE-BAEC-44569517829D"), ColumnValues = [new Guid("280D68FF-BEA9-47CE-BAEC-44569517829D")] },
+                new TestData { Item = new Guid("6AE44A0F-2345-49AA-95C0-ABB8E968E5CA"), ColumnValues = [new Guid("6AE44A0F-2345-49AA-95C0-ABB8E968E5CA")] },
+                new TestData { Item = new Guid("4F5D0D1E-EF63-4D3A-9519-585FF464477B"), ColumnValues = [new Guid("4F5D0D1E-EF63-4D3A-9519-585FF464477B")] }
+            ]
+        );
+    yield return ([new SqlMetaData("Decimal", SqlDbType.Decimal), new SqlMetaData("String", SqlDbType.VarChar, 10)],
+            [
+                new TestData { Item = 1, ColumnValues = [8.15m, "First"] },
+                new TestData { Item = 2, ColumnValues = [17.4m, "Second"] },
+                new TestData { Item = 3, ColumnValues = [23.7m, "Third"] }
+            ]
+        );
+  }
 
-    if(isDistinct)
-      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}{SqlTableValuedDataParameterDefinition.DistinctCollectionTableTypeNameSuffix}"));
-    else
-      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}"));
+  [Test]
+  [TestCaseSource(nameof(GetTestCollections))]
+  public void GetParameterValue_GetsValuesFromRecordDefinition_AndAddsThemToTvpValue ((SqlMetaData[] columnMetaData, TestData[] testData) testCase)
+  {
+    var columnMetaData = testCase.columnMetaData;
+    var testData = testCase.testData;
 
-    Assert.That(tvpValue.ColumnMetaData.Count, Is.EqualTo(1));
+    var storagePropertyDefinition = Mock.Of<IRdbmsStoragePropertyDefinition>();
 
-    var sqlMetaData = tvpValue.ColumnMetaData.Single();
-    Assert.That(sqlMetaData.Name, Is.EqualTo("Value"));
-    Assert.That(sqlMetaData.SqlDbType, Is.EqualTo(expectedSqlDbType));
+    var tableTypeDefinitionPartialMock = new Mock<TableTypeDefinition>(
+        new EntityNameDefinition("schema", "Test"),
+        new[] { storagePropertyDefinition },
+        Array.Empty<ITableConstraintDefinition>());
+    tableTypeDefinitionPartialMock.Setup(_ => _.CreateTableValuedParameterValue()).Returns(new SqlTableValuedParameterValue("schema.TestType", columnMetaData));
 
-    Assert.That(tvpValue.Select(record => record.GetValue(0)), Is.EqualTo(convertedValues));
-  }
+    var recordProperty = new FakeRecordPropertyDefinition(storagePropertyDefinition);
+    var recordDefinitionPartialMock = new Mock<RecordDefinition>("Test", tableTypeDefinitionPartialMock.Object, new[] { recordProperty });
+    foreach (var data in testData)
+      recordDefinitionPartialMock.Setup(_ => _.GetColumnValues(data.Item)).Returns(data.ColumnValues);
 
```

Comment:
`GetColumnValues()` returns the flattened list of the column values of all properties, and also converts them.
We would have to set up both `IRdbmsStoragePropertyDefinition.SplitValue()` for each item, and the returned `ColumnValue`s must each have a `ColumnDefinition` with an `IStorageTypeInformation`-Mock where we set up `ConvertToStorageType()` for each value.
This would make the test much more elaborate and less readable.

Reply:
We have the StorrageTypeInformation ObjectMother, so those wouldn't get mocks. Not sure about the IRdbmsStroagePropertyDefinition, how complicated the objectmother would be. My assumption is that we could simply set up test data using the real types instead of mocks. The test data we used didn't seem overly complex do I asumed it would not be much setup required when using real types. can you show how it would look like in 10-15 minutes?

Reply:
As discussed, I simplified the test so that no partial mock is necessary.